src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -970,6 +973,9 @@ static bool get_signal_code_description(const siginfo_t* si, enum_sigcode_desc_t
     { SIGFPE,  FPE_FLTSUB,   "FPE_FLTSUB",   "Subscript out of range." },
     { SIGSEGV, SEGV_MAPERR,  "SEGV_MAPERR",  "Address not mapped to object." },
     { SIGSEGV, SEGV_ACCERR,  "SEGV_ACCERR",  "Invalid permissions for mapped object." },
+#if defined(LINUX)
```

Comment:
```suggestion
#if defined(SEGV_BNDERR)
```
safer; in case other platforms have this too, and in case Linux has this not. As long as this is a macro, at least.

Reply:
I define it (line 54) if it is not there so it should be safe.

Reply:
Hmm, okay. 

That means that when we build on an older machine, and Linux kernels change the constant in newer kernel releases, we may use the wrong number.

What I would do is this:

```
#define SEGV_BNDERR_value 3
#if defined(SEGV_BNDERR)
STATIC_ASSERT(SEGV_BNDERR == SEGV_BNDERR_value)
#else
#define SEGV_BNDERR SEGV_BNDERR_value
#endif
```

The static assert will alert us if we build on a newer Linux version and turns out we were wrong with our assumed number.

Reply:
Not sure if this is very likely , but who knows.
I created  https://bugs.openjdk.org/browse/JDK-8353568

Reply:
Why a new bug, why not just do it as part of this PR? Am I missing something?

> Not sure if this is very likely , but who knows.

It can happen. With signal definitions I agree, very unlikely. But I had a bug last year with mmap constants (you'd think they never change once defined, too) where Oracle Linux changed those constants downstream in their kernel, and that caused errors at customers.